File: C_C++/beginners/some_sums.cpp
#include <iostream>
using namespace std;

//各桁の和
int findSumOfDigits(int n){
    int sum = 0;
    while(n > 0){
        sum += n % 10;
        n /= 10;
    }
    return sum;
}

int main(){
    int N, A, B;
    cin >> N >> A >> B;
    int total = 0;
    for(int i = 1; i <= N; i++){
        int sum = findSumOfDigits(i);
        if(A <= sum && sum <= B){
            total += i;
        }
    }
    cout << total << endl;
}

File: Python/beginners/some_sums.py
x, a, b = map(int, input().split())
if(1 <= x and x <= 10000) and (1 <= a and a <= b and b <= 36):
    total = 0
    for i in range(1, x+1):
        l = [int(x) for x in list(str(i))]
        s = sum(l)
        if(a <= s and s <= b):
            total += i
    print(total)